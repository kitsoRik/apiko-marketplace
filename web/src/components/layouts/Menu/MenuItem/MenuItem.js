import React from 'react';

import "./MenuItem.scss";

const MenuItem = ({ value = "", icon, _close, onAccept = () => { }, ...props }) => {

    const onClick = () => {
        onAccept();
        _close();
    }

    return (
        <div className="menu-item" onClick={onClick} {...props}>
            {icon}
            <span>{value}</span>
        </div>
    )
};

export default MenuItem;