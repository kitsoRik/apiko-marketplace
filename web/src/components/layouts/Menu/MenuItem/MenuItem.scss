.menu-item {
	padding: 5px 10px;
	height: 20px;

	display: grid;
	grid-template-columns: 24px auto;
	grid-column-gap: 10px;

	font-size: 18px;
	align-items: center;

	cursor: pointer;

	&:hover {
		background-color: #eee;
	}
}
