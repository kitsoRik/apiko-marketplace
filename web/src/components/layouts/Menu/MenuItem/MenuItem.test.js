import React from "react";
import { create } from "react-test-renderer";
import MenuItem from "./MenuItem";
import { mount } from "enzyme";
describe("MenuItem", () => {
    it("should be rendered", () => {
        const component = create(<MenuItem />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with value", () => {
        const component = create(<MenuItem value="Custom value" />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with accept event", () => {
        const component = create(<MenuItem onAccept={() => { }} />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should emit accept event", () => {
        const onAccept = jest.fn();
        const component = mount(<MenuItem _close={() => { }} onAccept={onAccept} />);
        const div = component.find("div");

        div.simulate("click");
        expect(onAccept).toBeCalledTimes(1);
        div.simulate("click");
        expect(onAccept).toBeCalledTimes(2);
    });
});
