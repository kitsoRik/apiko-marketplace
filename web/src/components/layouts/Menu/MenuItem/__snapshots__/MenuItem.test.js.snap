// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should be rendered 1`] = `
<div
  className="menu-item"
  onClick={[Function]}
>
  <span>
    
  </span>
</div>
`;

exports[`MenuItem should render with accept event 1`] = `
<div
  className="menu-item"
  onClick={[Function]}
>
  <span>
    
  </span>
</div>
`;

exports[`MenuItem should render with value 1`] = `
<div
  className="menu-item"
  onClick={[Function]}
>
  <span>
    Custom value
  </span>
</div>
`;
